### ISSUE TITLE
onClick" was passed to <Link> with `href`

### ISSUE THREAD

**Issue by mschinis** [amannn/next-intl#1271]
### Description

I'm trying to use shadcn NavigationMenu together with the next-intl link, using legacyBehavior+passHref but I'm getting a bunch of the following warnings:

```
"onClick" was passed to <Link> with `href` of `/about` but "legacyBehavior" was set. The legacy behavior requires onClick be set on the child of next/link 
```

**Usage**
```tsx
// Import next-intl Link component returned by createSharedPathnamesNavigation
import { Link } from "@/i18n/navigation"; 

function NavigationMenuLocalizedLink({ item }: { item: INavigationMenuItem }) {
  return (
    <Link href={item.href} legacyBehavior passHref>
      <NavigationMenuLink>{item.title}</NavigationMenuLink>
    </Link>
  );
}
```

Tracing back the issue, it looks like nextjs doesn't like setting the `onClick` property, when `legacyBehavior` is set.
https://github.com/amannn/next-intl/blob/ed4681fa70b8f67a97e1364deaabf58df302d80c/packages/next-intl/src/navigation/shared/BaseLink.tsx#L76

### Verifications

- [X] I've verified that the problem I'm experiencing isn't covered in [the docs](https://next-intl-docs.vercel.app/).
- [X] I've searched for similar, existing issues on GitHub and [Stack Overflow](https://stackoverflow.com/search?q=next-intl).
- [X] I've compared my app to [a working example](https://next-intl-docs.vercel.app/examples) to look for differences.

### Mandatory reproduction URL

https://github.com/mschinis/next-intl-bug-repro-app-router

### Reproduction description

Steps to reproduce:
1. Open reproduction
2. Look at console

### Expected behaviour

Expected behavior is to not have any warnings.

**Comment by mschinis** [comment]
The following workaround seems to work for now, but the behavior of the issue still stands:

```tsx
function NavigationMenuLocalizedLink({ item }: { item: INavigationMenuItem }) {
  return (
    <NavigationMenuLink asChild>
      <Link href={item.href}>{item.title}</Link>
    </NavigationMenuLink>
  );
}

```

**Comment by amannn** [comment]
Thanks for the careful report and reproduction!

This could probably be addressed by adding `cloneElement` in `BaseLink` in case `legacyBehavior` is detected. I'm not sure though if it's worth it though, given that—as the name suggests—this is legacy behavior and there are modern alternatives (like [the one you suggested](https://github.com/amannn/next-intl/issues/1271#issuecomment-2296436156)).

This was in fact the first time this was reported, so maybe it's a good idea to keep this open to see if more people are affected and to give visibility to the alternative you've posted.

**Comment by rogerogers** [comment]
Yes, I'm also facing the same issue. Directly importing from `next/link` does not have this problem.

**Comment by amannn** [comment]
This topic came up again in https://github.com/amannn/next-intl/issues/1322 and I've added some details in https://github.com/amannn/next-intl/issues/1322#issuecomment-2329841708.

A non-obtrusive workaround might be this:

```tsx
// components/Link.tsx
'use client';

// Re-export from the file that creates the `<Link />` component
export {Link as default} from '@/routing';
```

```tsx
// Import the re-exported `Link`
import Link from "components/Link"; 

function NavigationMenuLocalizedLink({ item }: { item: INavigationMenuItem }) {
  return (
    <Link href={item.href} legacyBehavior passHref>
      <NavigationMenuLink>{item.title}</NavigationMenuLink>
    </Link>
  );
}
```

I think this should work as expected with Radix Primitives & friends.

**EDIT:** Asked the Radix team here about an alternative implementation that would work with RSC: https://github.com/radix-ui/primitives/issues/2537#issuecomment-2381324518.

**Comment by amannn** [comment]
It seems like Radix Primitives will switch to a new API eventually that does not require `cloneElement`: https://x.com/vladyslavmoroz/status/1863986795601363285

**Comment by LikeDreamwalker** [comment]
Any updates here?

**Comment by escwxyz** [comment]
Looking for updates

**Comment by amannn** [comment]
A workaround is mentioned above: https://github.com/amannn/next-intl/issues/1271#issuecomment-2331863819

**Comment by bennajah** [comment]
Some shadcn/ui components don’t work correctly when using asChild with Link (for example: Button, NavigationMenuLink).
However, they work as expected when using the default Link from next/link.

**Comment by amannn** [comment]
Interesting experiment: https://github.com/jjenzz/slot-jsx-pragma ([X post](https://x.com/jjenzz/status/1997258067587797390)). Seems like the Radix team is still interested in solving this …

**Comment by Lukapetro** [comment]
I ran into this issue and did some debugging. Here's what I found:

**Root cause:** When using the `render` prop with Next.js `Link` inside a **Server Component**, Next.js App Router automatically wraps client components in `React.lazy()`. The `render` prop ends up receiving a lazy object (`Symbol(react.lazy)`) instead of a resolved React element, which Base UI then passes directly to React — causing the "Element type is invalid" error.

```
// ❌ Breaks in Server Components (no "use client")
<Button render={<Link href="/foo" />}>Click me</Button>
```

**Workaround:** Wrap the Button + Link composition in a Client Component:
```
"use client";

import { Button } from "@base-ui/react/button";
import Link from "next/link";

export function LinkButton({ href, children, ...props }) {
  return (
    <Button render={<Link href={href} />} {...props}>
      {children}
    </Button>
  );
}
```
This ensures the `Link` is resolved before being passed to the `render` prop.
